NUsenseCpp/USB_DEVICE/App/usbd_cdc_if.c

```diff
@@ -263,9 +267,21 @@ static int8_t CDC_Control_HS(uint8_t cmd, uint8_t* pbuf, uint16_t length)
   */
 static int8_t CDC_Receive_HS(uint8_t* Buf, uint32_t *Len)
 {
-  /* USER CODE BEGIN 11 */
-  USBD_CDC_SetRxBuffer(&hUsbDeviceHS, &Buf[0]);
-  USBD_CDC_ReceivePacket(&hUsbDeviceHS);
+	/* USER CODE BEGIN 11 */
+	USBD_CDC_SetRxBuffer(&hUsbDeviceHS, &Buf[0]);
+
+	// Raise flags
+	rx_flag = 1;
+	rx_len = *Len;
+
+	// Do memcpy here to append data from the supplied buffer to rx_buf
+	memcpy(&rx_buf[rx_buf_len], &Buf[0], rx_len);
+
+	rx_buf_len = rx_buf_len + rx_len;

```

Comment:
```suggestion
	rx_buf_len = (rx_buf_len + rx_len) % RX_BUF_SIZE;
```

Make sure to wrap around so we dont get a segfault later. This variable just notes the element after the last byte in the buffer so this should be fine

Reply:
👍